packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js

```diff
@@ -17,17 +23,54 @@ import AddFilterCTA from '../../components/AddFilterCTA';
 const HomePage = () => {
   const { formatMessage } = useGlobalContext();
   const [reducerState, dispatch] = useReducer(reducer, initialState, init);
-  const [isOpen, setIsOpen] = useState(true);
-  const { data, dataToDelete, _q } = reducerState.toJS();
+  const [isOpen, setIsOpen] = useState(false);
+  const { push } = useHistory();
+  const query = useQuery();
+  const { data, dataToDelete } = reducerState.toJS();
   const pluginName = formatMessage({ id: getTrad('plugin.name') });
 
-  const handleClickToggleModal = () => {
-    setIsOpen(prev => !prev);
-  };
-  const handleClearSearch = () => {
+  useEffect(() => {
+    // TODO - Retrieve data
     dispatch({
-      type: 'ON_CLEAR_SEARCH',
+      type: 'GET_DATA_SUCCEEDED',
+      data: [],
+    });
+  }, []);
+
+  const getSearchParams = () => {
+    const params = {};
+    query.forEach((value, key) => {
+      params[key] = value;
     });
+
+    return params;
+  };
+
+  const getUpdatedSearchParams = updatedParams => {
+    return {
+      ...getSearchParams(),
+      ...updatedParams,
+    };
+  };
+
+  const getQueryValue = key => {
+    const queryParams = getSearchParams();
+
+    return queryParams[key];
+  };
+
+  const handleChangeParams = ({ target: { name, value } }) => {
+    const updatedSearch = getUpdatedSearchParams({ [name]: value });
+    const newSearch = generateSearchFromFilters(updatedSearch);
```

Comment:
Here you might want to use `encodeURIComponent` but if you do so you will need to update the utilities from the `strapi-helper-plugin`

Reply:
Can we use `encodeURI` instead of `encodeURIComponent`? 
To encode special characters, except: `, / ? : @ & = + $ #`
Or do we want to encore them too ?

Reply:
`encodeURI` and we will update the utility later, good catch!